Sources/Build/BuildPlan.swift

```diff
@@ -1062,6 +1062,15 @@ public final class SwiftTargetBuildDescription {
         // Swift defines.
         let swiftDefines = scope.evaluate(.SWIFT_ACTIVE_COMPILATION_CONDITIONS)
         flags += swiftDefines.map({ "-D" + $0 })
+        
+        // As a special case, if `SWIFT_ENABLE_REGEX_LITERALS` we also pass `-enable-regex-literals` if the compiler
+        // supports it.
+        let canEnableRegularExpressionLiterals = SwiftTargetBuildDescription.checkSupportedFrontendFlags(
+            flags: ["enable-regex-literals"], fileSystem: self.fileSystem
+        )
+        if swiftDefines.contains("SWIFT_ENABLE_REGEX_LITERALS") && canEnableRegularExpressionLiterals {
+            flags += ["-enable-regex-literals"]
+        }
```

Comment:
should the order of the conditions be reversed so that the more expensive check (driver based) is only called did the less expensive one (setting based) is set?

Reply:
Yeah, that's a good point.  I had assumed it was cached but then looking into how `checkSupportedFrontendFlags` is actually implemented it looks pretty expensive.

I think I will probably want another go at making this a cleaner implementation.